add function & menu item to delete book

ewarlock opened the pull request:
Added a function in main.py that prompts user to enter book id. It looks up a book by id, then deletes that book. If an invalid id is entered, an error message prints: book not found. Added menu item 7 for user to select that calls this function. For issue #2  @3bdifatah

ewarlock committed:
add function & menu item to delete book

3bdifatah commented:
great feature added, it works fine three small suggestions I would add is:
1. add empty line after error message to make the error readable to the user on line 76.
2. confirm the user if they want to delete f'{book_id} by: {author}' before deleting the book on line 74.
3. show all the books to the user so they know the ID of the book before entering book_id on line 72. not necessary but saves time for the user.

claraj commented:
> great feature added, it works fine three small suggestions I would add is:

These two sound good to me. For 2. where are all the input statements in the code? If you make this change follow the current organization.  

> 1. add empty line after error message to make the error readable to the user on line 76.
> 2. confirm the user if they want to delete f'{book_id} by: {author}' before deleting the book on line 74.

And this one is probably optional - there's another menu option to display all the books so user can look up their ID that way, so agree not necessary. But, I'm happy if you implement it - what do you both think?

> 3. show all the books to the user so they know the ID of the book before entering book_id on line 72. not necessary but saves time for the user.

3bdifatah commented:
all the input statements will go into ui.py file. @ewarlock and I will discuss if we should implement the 3rd or not.

ewarlock committed:
reformat error message to be more user friendly

ewarlock committed:
add book delete confirmation to ui.py & main.py delete_book()

ewarlock committed:
add documentation for ui.py confirm_deletion

ewarlock committed:
define specific exception in main.py delete_book UnboundLocalError

ewarlock commented:
Modified the error message to be more user friendly with new lines and listing the book ID. I also modified the code in main.py to utilize some of the functions in ui.py. I added a function in ui.py confirm_deletion so the user can confirm that they really want to delete the book. I also specified the error that is being caught in delete_book. 

Let me know what you think @3bdifatah  ! I did not implement 3 since the user can look at their list of books anytime from main menu (and we haven't implemented it in other areas where the user might have a book ID related error.)

3bdifatah commented:
works great and liked how the user is taken back to the main menu if they decide not to delete and also prints friendly message as well. I think it looks great and you can merge it.

claraj commented on main.py:
```diff
@@ -64,6 +65,20 @@ def change_read():
+            return
+
+        book.delete()
+    except UnboundLocalError:
```
UnboundLocalError is one that you very rarely need to catch. If you get this error in your program, it's almost always a coding error, and it's the same error that was fixed in PR #11.   Incorporating changes from #11 you would try to find the book, test if the book is None, if so, it doesn't exist, otherwise it does exist and you can delete it.
